src/lib/datepicker/calendar.ts

```diff
@@ -95,7 +107,8 @@ export class MdCalendar<D> implements AfterContentInit {
   /** The label for the current calendar view. */
   get _periodButtonText(): string {
     return this._monthView ?
-        this._dateAdapter.getMonthYearName(this._activeDate, 'short').toLocaleUpperCase() :
+        this._dateAdapter.format(this._activeDate, this._dateFormats.display.monthYearLabel)
+            .toLocaleUpperCase() :
```

Comment:
I don't think either the format language used by Intl.DatetimeFormat or Moment.js gives control over the case, so we're stuck doing this